crates/sui-core/src/node_sync/node_follower.rs

```diff
@@ -632,6 +635,33 @@ where
     }
 }
 
+pub struct MappedRx(oneshot::Receiver<SuiResult>);
+
+impl Future for MappedRx {
```

Comment:
@bmwill please let me know if you know a more concise way to do this - this would normally just be a `.map` call via FuturesExt, but I need a nameable future to use as the type for the trait below.